base/applications/calc/utl_ieee.c

```diff
@@ -40,11 +40,12 @@ void prepare_rpn_result_2(calc_number_t *rpn, TCHAR *buffer, int size, int base)
         /* calculate the width of integer number */
         width = (rpn->f==0) ? 1 : (int)log10(fabs(rpn->f))+1;
         if (calc.sci_out == TRUE || width > MAX_LD_WIDTH || width < -MAX_LD_WIDTH)
-            _stprintf(buffer, _T("%#.*e"), MAX_LD_WIDTH-1, rpn->f);
+            StringCchPrintf(buffer, size, _T("%#.*e"), MAX_LD_WIDTH-1, rpn->f);
         else {
             TCHAR *ptr, *dst;
 
-            ptr = buffer + _stprintf(buffer, _T("%#*.*f"), width, ((MAX_LD_WIDTH-width-1)>=0) ? MAX_LD_WIDTH-width-1 : 0, rpn->f);
+            StringCchPrintf(buffer, size, _T("%#*.*f"), width, ((MAX_LD_WIDTH-width-1)>=0) ? MAX_LD_WIDTH-width-1 : 0, rpn->f);
+            ptr = buffer + _tcslen(buffer);
```

Comment:
What about:
```suggestion
            StringCchPrintfEx(buffer, size, &ptr, NULL, 0, _T("%#*.*f"), width, ((MAX_LD_WIDTH-width-1)>=0) ? MAX_LD_WIDTH-width-1 : 0, rpn->f);
```
Note also I've put "0" for the flags, maybe another one may be better?

Reply:
```txt
[in] dwFlags
Type: DWORD
One or more of the following values.
```
It's more than zero.

Reply:
Not really. You can pass the value 0 in case you don't want any of the behaviours selected by the non-zero flags.

Reply:
Fixed.